### PR TITLE
API-410: Add create family variant and product model

### PR DESCRIPTION
**Pull request by willy-ahva**

### CODE REVIEW

File: src/Api/ProductModelApi.php [hunk 1]
```diff
@@ -35,4 +36,14 @@ public function get($code)
     {
         return $this->resourceClient->getResource(static::PRODUCT_MODEL_URI, [$code]);
     }
+
+    /**
+     * Available from Akeneo PIM 2.0.
+     *
+     * {@inheritdoc}
+     */
+    public function create(array $data = [])
```
Comment (skeleton) [inline src/Api/ProductModelApi.php]: same remark

File: src/Client/HttpExceptionHandler.php [hunk 2]
```diff
@@ -79,6 +79,21 @@ protected function getResponseMessage(ResponseInterface $response)
         $decodedBody = json_decode($responseBody->getContents(), true);
         $responseBody->rewind();
 
-        return isset($decodedBody['message']) ? $decodedBody['message'] : $response->getReasonPhrase();
+        $message = isset($decodedBody['message']) ? $decodedBody['message'] : $response->getReasonPhrase();
```
Comment (skeleton) [inline src/Client/HttpExceptionHandler.php]: there is already a way to get error messages (see https://api.akeneo.com/php-client/exception.html#unprocessable-entity-exception). do you really think it's needed to change this behavior? And if true, update the doc please

File: tests/v2_0/Api/FamilyVariant/CreateFamilyVariantApiIntegration.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Akeneo\Pim\tests\v2_0\Api\FamilyVariant;
+
+use Akeneo\Pim\tests\Common\Api\ApiTestCase;
+
+class CreateFamilyVariantApiIntegration extends ApiTestCase
+{
+    public function testCreateAFamilyVariant()
```
Comment (skeleton) [inline tests/v2_0/Api/FamilyVariant/CreateFamilyVariantApiIntegration.php]: we should add test when the creation returns errors (422, validation error)

File: tests/v2_0/Api/ProductModel/CreateProductModelIntegration.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Akeneo\Pim\tests\v2_0\Api\ProductModel;
+
+use Akeneo\Pim\tests\Common\Api\ApiTestCase;
+
+class CreateProductModelIntegration extends ApiTestCase
+{
+    public function testCreateAProductModel()
```
Comment (skeleton) [inline tests/v2_0/Api/ProductModel/CreateProductModelIntegration.php]: same remark

File: src/Client/HttpExceptionHandler.php (hunk 2 above)
Comment (willy-ahva) [reply src/Client/HttpExceptionHandler.php]: ah
It's not needed anymore :p

File: tests/v2_0/Api/FamilyVariant/CreateFamilyVariantApiIntegration.php (hunk 3 above)
Comment (willy-ahva) [reply tests/v2_0/Api/FamilyVariant/CreateFamilyVariantApiIntegration.php]: done

File: tests/v2_0/Api/ProductModel/CreateProductModelIntegration.php (hunk 4 above)
Comment (willy-ahva) [reply tests/v2_0/Api/ProductModel/CreateProductModelIntegration.php]: done

**Pull request closed without merging**